global_costmap:
  map_topic: map #map_for_costmap
  global_frame: map
  robot_base_frame: base_footprint
  update_frequency: 3.0
  publish_frequency: 3.0
  transform_tolerance: 0.5
  always_send_full_costmap: false
  rolling_window: false
  resolution: 0.1
  
  plugins:
    - {name: static_layer, type: "costmap_2d::StaticLayer"}
    - {name: obstacle_layer, type: "costmap_2d::ObstacleLayer"}
    - {name: obstacle_layer2, type: "costmap_2d::ObstacleLayer"}
    - {name: inflater_layer, type: "costmap_2d::InflationLayer"}
  
  static_layer:
    map_topic: map
    subscribe_to_updates: true
    lethal_cost_threshold: 50
    track_unknown_space: false
  
  obstacle_layer:
    observation_sources: scan 
    scan:
      data_type: LaserScan
      topic: /hokuyo_scan
      sensor_frame: hokuyo_link
      marking: true
      clearing: true
      inf_is_valid: true
      max_obstacle_height: 0.3
      min_obstacle_height: 0
      obstacle_range: 3
      raytrace_range: 3
      observation_persistence: 0.0
      
  obstacle_layer2:
    observation_sources: scan 
    scan:
      data_type: LaserScan
      topic: /velodyne_scan
      sensor_frame: velodyne_link
      marking: true
      clearing: true
      inf_is_valid: true
      max_obstacle_height: 2.5
      min_obstacle_height: 0
      obstacle_range: 10
      raytrace_range: 10
      observation_persistence: 0.0
      
  inflater_layer:
    inflation_radius: 5.0
    cost_scaling_factor: 1.5


local_costmap:
  global_frame: map
  robot_base_frame: base_footprint
  update_frequency: 6.0
  publish_frequency: 5.0
  rolling_window: true
  resolution: 0.05
  width: 8.0
  height: 8.0
  transform_tolerance: 0.5
  always_send_full_costmap: true
  
  plugins:
    - {name: obstacle_layer1, type: "costmap_2d::ObstacleLayer"}
    - {name: obstacle_layer2, type: "costmap_2d::ObstacleLayer"}
    - {name: inflater_layer, type: "costmap_2d::InflationLayer"}
  
  obstacle_layer1:
    observation_sources: scan 
    scan:
      data_type: LaserScan
      topic: /scan
      sensor_frame: velodyne_link
      marking: true
      clearing: true
      inf_is_valid: true
      max_obstacle_height: 2.5
      min_obstacle_height: -0.5
      obstacle_range: 10
      raytrace_range: 10
      observation_persistence: 0.0
  
  inflater_layer:
    inflation_radius: 0.4
    cost_scaling_factor: 5.0

# https://answers.ros.org/question/326867/local_costmap-not-showing-every-obstacle/


base_local_planner: base_local_planner/TrajectoryPlannerROS

TrajectoryPlannerROS:
  # Robot Configuration Parameters
  holonomic_robot: false
  acc_lim_x: 3.5 #4.5
  acc_lim_theta: 1.2 #2.0
  max_vel_x: 1.2
  min_vel_x: 0.0 #0.1
  max_vel_theta: 1.0
  min_vel_theta: -1.0
  min_in_place_vel_theta: 0.2
  escape_vel: -0.3
  
  # Goal Tolerance Parameters
  yaw_goal_tolerance: 0.6 #0.3
  xy_goal_tolerance: 0.3
  
  # Forward Simulation Parameters
  sim_time: 2.0
  sim_granularity: 0.05
  vx_samples: 3
  vtheta_samples: 20
  
  # Trajectory Scoring Parameters
  meter_scoring: true
  dwa: false #true
  path_distance_bias: 1.0 #0.8 #0.4
  goal_distance_bias: 0.6 #0.9
  occdist_scale : 0.01 #0.05
  heading_lookahead: 0.325
  heading_scoring: true #
  heading_scoring_timestep: 1.0
